Easydict/NewApp/View/SettingView/Tabs/ServiceTab.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  ServiceTab.swift
+//  Easydict
+//
+//  Created by phlpsong on 2024/1/6.
+//  Copyright © 2024 izual. All rights reserved.
+//
+
+import SwiftUI
+
+@available(macOS 13, *)
+struct ServiceTab: View {
+    @State private var windowTypeValue = EZWindowType.mini.rawValue
+    @State private var serviceTypes: [ServiceType] = []
+    @State private var services: [QueryService] = []
+    @State private var selectedIndex: Int?
+
+    var segmentCtrl: some View {
+        Picker("", selection: $windowTypeValue) {
+            Text("mini_window")
+                .tag(EZWindowType.mini.rawValue)
+
+            Text("fixed_window")
+                .tag(EZWindowType.fixed.rawValue)
+
+            Text("main_window")
+                .tag(EZWindowType.main.rawValue)
+        }
+        .padding()
+        .pickerStyle(.segmented)
+        .onChange(of: windowTypeValue) { type in
+            loadService(type: type)
+        }
+    }
+
+    var serviceList: some View {
+        List {
+            ForEach(Array(serviceTypes.enumerated()), id: \.offset) { index, _ in
+                ServiceItemView(
+                    service: $services[index]
```

Comment:
打开设置，切换到服务页就崩溃了，提示数组越界，debug 发现是这里有问题。

```
*** -[__NSFrozenArrayM objectAtIndex:]: index 13 beyond bounds [0 .. 12]
(
	0   CoreFoundation                      0x000000018fa76800 __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x000000018f56deb4 objc_exception_throw + 60
	2   CoreFoundation                      0x000000018f9e8be0 _CFURLGetObjectInformationNoIO + 0
	3   libswiftCore.dylib                  0x000000019f1a9250 $ss12_ArrayBufferV19_getElementSlowPathyyXlSiF + 196
	4   libswiftCore.dylib                  0x000000019f1b02e8 $sSayxSicir + 304
	5   libswiftCore.dylib                  0x000000019f1b0184 $sSayxGSlsSly7ElementQz5IndexQzcirTW + 68
	6   SwiftUI                             0x00000001bbaee600 OUTLINED_FUNCTION_24 + 6872
	7   SwiftUI                             0x00000001bbaed0d4 OUTLINED_FUNCTION_24 + 1452
	8   SwiftUI                             0x00000001bbaee518 OUTLINED_FUNCTION_24 + 6640
	9   Easydict-debug                      0x0000000104a27480 $s8Easydict10ServiceTabV11serviceListQrvg7SwiftUI4ViewPAEE16listRowSeparator_5edgesQrAE10VisibilityO_AE12VerticalEdgeO3SetVtFQOyAE07DynamicH7ContentPAEE6onMove7performQry10Foundation05IndexP0V_SitcSg_tFQOyAE7ForEachVySaySi6offset_So13EZServiceTypea7elementtGSiAgEE0iJ10BackgroundyQrqd__SgAeFRd__lFQOyAgEE0S10TapGesture5countASQrSi_yyctFQOyAgEE12contentShape_6eoFillQrqd___SbtAE5ShapeRd__lFQOyAgEE3tagyQrqd__SHRd__lFQOyAA0b4ItemH0V_SiQo__AE9RectangleVQo__Qo__AE5ColorVQo_G_Qo__Qo_yXEfU_A20_Si_A0_tcfU_ + 632
	10  Easydict-debug                      0x0000000104a2c1f8 $s8Easydict10ServiceTabV11serviceListQrvg7SwiftUI4ViewPAEE16listRowSeparator_5edgesQrAE10VisibilityO_AE12VerticalEdgeO3SetVtFQOyAE07DynamicH7ContentPAEE6onMove7performQry10Foundation05IndexP0V_SitcSg_tFQOyAE7ForEachVySaySi6offset_So13EZServiceTypea7elementtGSiAgEE0iJ10BackgroundyQrqd__SgAeFRd__lFQOyAgEE0S10TapGesture5countASQrSi_yyctFQOyAgEE12contentShape_6eoFillQrqd___SbtAE5ShapeRd__lFQOyAgEE3tagyQrqd__SHRd__lFQOyAA0b4ItemH0V_SiQo__AE9RectangleVQo__Qo__AE5ColorVQo_G_Qo__Qo_yXEfU_A20_Si_A0_tcfU_TA + 16
	11  Easydict-debug                      0x0000000104a28188 $sSiSo13EZServiceTypea7SwiftUI15ModifiedContentVyAC4ViewPACE12onTapGesture5count7performQrSi_yyctFQOyAEyAEy8Easydict011ServiceItemG0VAC21_TraitWritingModifierVyAC08TagValueP3KeyVySiGGGAC01_f5ShapeR0VyAC9RectangleVGG_Qo_AOyAC017ListRowBackgroundpU0VGGIegygr_Si6offset_AB7elementtA3_Iegnr_TR + 36
	12  Easydict-debug                      0x0000000104a2c250 $sSiSo13EZServiceTypea7SwiftUI15ModifiedContentVyAC4ViewPACE12onTapGesture5count7performQrSi_yyctFQOyAEyAEy8Easydict011ServiceItemG0VAC21_TraitWritingModifierVyAC08TagValueP3KeyVySiGGGAC01_f5ShapeR0VyAC9RectangleVGG_Qo_AOyAC017ListRowBackgroundpU0VGGIegygr_Si6offset_AB7elementtA3_Iegnr_TRTA + 20
	13  SwiftUI                             0x00000001bb4ece84 OUTLINED_FUNCTION_52 + 18912
	14  SwiftUI                             0x00000001bb4f7734 OUTLINED_FUNCTION_52 + 62096
	15  SwiftUI                             0x00000001babe918c OUTLINED_FUNCTION_58 + 56
	16  SwiftUI                             0x00000001bb4f7754 OUTLINED_FUNCTION_52 + 62128
	17  libswiftCore.dylib                  0x000000019f1df6a4 $ss24withUnsafeMutablePointer2to_q_xz_q_SpyxGKXEtKr0_lF + 28
	18  SwiftUI                             0x00000001bb4ec25c OUTLINED_FUNCTION_52 + 15800
	19  SwiftUI                             0x00000001bb4eeabc OUTLINED_FUNCTION_52 + 26136
	20  SwiftUI                             0x00000001bb4f04d0 OUTLINED_FUNCTION_52 + 32812
	21  SwiftUI                             0x00000001bb4f4c84 OUTLINED_FUNCTION_52 + 51168
	22  SwiftUI                             0x00000001bb4f50c0 OUTLINED_FUNCTION_52 + 52252
	23  SwiftUI                             0x00000001bbcffba8 OUTLINED_FUNCTION_18 + 13980
	24  SwiftUI                             0x00000001bbcfc964 OUTLINED_FUNCTION_18 + 1112
	25  SwiftUI                             0x00000001bac1425c OUTLINED_FUNCTION_3 + 5196
	26  SwiftUI                             0x00000001bac15418 OUTLINED_FUNCTION_3 + 9736
	27  SwiftUI                             0x00000001bac1561c OUTLINED_FUNCTION_3 + 10252
	28  SwiftUI                             0x00000001ba9e3fbc OUTLINED_FUNCTION_1 + 3692
	29  SwiftUI                             0x00000001baa200f0 OUTLINED_FUNCTION_64 + 5120
	30  SwiftUI                             0x00000001baa20178 OUTLINED_FUNCTION_64 + 5256
	31  AppKit                              0x0000000193309b60 -[NSTableView _delegate_isGroupRow:] + 80
	32  AppKit                              0x0000000193d9c3e8 -[NSTableRowData _cacheGroupRowIndexesWithinRange:] + 100
	33  AppKit                              0x0000000193d9c340 -[NSTableRowData isGroupRowIndex:] + 144
	34  AppKit                              0x0000000193896250 -[NSTableView _spacingPrecedingRow:] + 44
	35  AppKit                              0x0000000193d10fbc -[NSTableRowHeightData _cacheRowSpansInRange:heightProvider:] + 588
	36  AppKit                              0x0000000193d0f534 -[NSTableRowHeightData _cacheRowSpansInRange:] + 100
	37  Foundation                          0x0000000190ae5570 __NSINDEXSET_IS_CALLING_OUT_TO_A_RANGE_BLOCK__ + 24
	38  Foundation                          0x00000001911faa94 __NSIndexSetEnumerateBitfield + 408
	39  AppKit                              0x0000000193d111fc -[NSTableRowHeightData _estimatedSpanForRow:cacheIfNecessary:] + 256
	40  AppKit                              0x0000000193d0e6e4 -[NSTableRowHeightData _estimatedSpanForRowsInRange:] + 48
	41  AppKit                              0x0000000193d0e818 -[NSTableRowHeightData computeTotalRowsSpan] + 68
	42  AppKit                              0x0000000193d0ea24 -[NSTableRowHeightData _estimatedRowAtOffset:] + 36
	43  AppKit                              0x0000000193263384 -[NSTableView rowAtPoint:] + 204
	44  AppKit                              0x0000000193d9a360 -[NSTableRowData _keepTopRowStableAtLeastOnce:andDoWorkUntilDone:] + 192
	45  AppKit                              0x0000000193271fc8 -[NSTableRowData _updateVisibleViewsBasedOnUpdateItems] + 1300
	46  AppKit                              0x000000019325a780 -[NSTableRowData endUpdates] + 328
	47  AppKit                              0x000000019325a5f4 -[NSTableView _endUpdateWithTile:] + 108
	48  SwiftUI                             0x00000001baa1f324 OUTLINED_FUNCTION_64 + 1588
	49  SwiftUI                             0x00000001bb3d53cc OUTLINED_FUNCTION_23 + 65416
	50  SwiftUI                             0x00000001bb3d4570 OUTLINED_FUNCTION_23 + 61740
	51  SwiftUI                             0x00000001bb3d39e8 OUTLINED_FUNCTION_23 + 58788
	52  SwiftUI                             0x00000001bb3d2418 OUTLINED_FUNCTION_23 + 53204
	53  SwiftUI                             0x00000001baaa6414 OUTLINED_FUNCTION_0 + 16392
	54  SwiftUI                             0x00000001baaab64c __swift_memcpy216_8 + 904
	55  SwiftUI                             0x00000001bb8cf4b0 OUTLINED_FUNCTION_1 + 9376
	56  SwiftUI                             0x00000001baaa60e0 OUTLINED_FUNCTION_0 + 15572
	57  SwiftUI                             0x00000001bb544888 __swift_allocate_boxed_opaque_existential_1Tm + 268
	58  AttributeGraph                      0x00000001bc071790 _ZN2AG5Graph11UpdateStack6updateEv + 512
	59  AttributeGraph                      0x00000001bc071fcc _ZN2AG5Graph16update_attributeENS_4data3ptrINS_4NodeEEEj + 424
	60  AttributeGraph                      0x00000001bc080814 _ZN2AG8Subgraph6updateEj + 848
	61  SwiftUI                             0x00000001bb97e5a4 OUTLINED_FUNCTION_8 + 12300
	62  SwiftUI                             0x00000001bba539cc OUTLINED_FUNCTION_2 + 21436
	63  SwiftUI                             0x00000001bba50670 OUTLINED_FUNCTION_2 + 8288
	64  SwiftUI                             0x00000001bb0da194 OUTLINED_FUNCTION_0 + 59928
	65  SwiftUI                             0x00000001bb0e5768 OUTLINED_FUNCTION_0 + 106476
	66  SwiftUI                             0x00000001bb0da628 OUTLINED_FUNCTION_0 + 61100
	67  AppKit                              0x00000001931f1a98 +[NSAnimationContext runAnimationGroup:] + 56
	68  SwiftUI                             0x00000001bb0da070 OUTLINED_FUNCTION_0 + 59636
	69  SwiftUI                             0x00000001bb0da668 OUTLINED_FUNCTION_0 + 61164
	70  AppKit                              0x0000000193c686b0 ___NSViewLayout_block_invoke + 592
	71  AppKit                              0x000000019321cab0 NSPerformVisuallyAtomicChange + 108
	72  AppKit                              0x00000001932215e0 _NSViewLayout + 96
	73  AppKit                              0x0000000193c5ec88 __36-[NSView _layoutSubtreeWithOldSize:]_block_invoke + 364
	74  AppKit                              0x000000019321cab0 NSPerformVisuallyAtomicChange + 108
	75  AppKit                              0x0000000193221574 -[NSView _layoutSubtreeWithOldSize:] + 100
	76  AppKit                              0x0000000193c5edcc __36-[NSView _layoutSubtreeWithOldSize:]_block_invoke + 688
	77  AppKit                              0x000000019321cab0 NSPerformVisuallyAtomicChange + 108
	78  AppKit                              0x0000000193221574 -[NSView _layoutSubtreeWithOldSize:] + 100
	79  AppKit                              0x0000000193c5f770 __56-[NSView _layoutSubtreeIfNeededAndAllowTemporaryEngine:]_block_invoke + 796
	80  AppKit                              0x000000019321cab0 NSPerformVisuallyAtomicChange + 108
	81  AppKit                              0x0000000193221134 -[NSView _layoutSubtreeIfNeededAndAllowTemporaryEngine:] + 100
	82  AppKit                              0x000000019321cab0 NSPerformVisuallyAtomicChange + 108
	83  AppKit                              0x00000001932210c4 -[NSView layoutSubtreeIfNeeded] + 96
	84  AppKit                              0x0000000193f0a528 -[NSWindow(NSConstraintBasedLayoutInternal) _layoutViewTree] + 104
	85  AppKit                              0x0000000193f0a6b0 -[NSWindow(NSConstraintBasedLayoutInternal) layoutIfNeeded] + 240
	86  AppKit                              0x00000001932d80b0 -[NSWindow _setUpFirstResponderBeforeBecomingVisible] + 68
	87  AppKit                              0x00000001932d7598 -[NSWindow _doWindowWillBeVisibleAsSheet:] + 108
	88  AppKit                              0x0000000193c7bd14 -[NSWindow _reallyDoOrderWindowAboveOrBelow:] + 1028
	89  AppKit                              0x0000000193c7c8c4 -[NSWindow _reallyDoOrderWindow:] + 64
	90  AppKit                              0x0000000193c7cb24 -[NSWindow _doOrderWindow:] + 264
	91  AppKit                              0x00000001932cd774 -[NSWindow makeKeyAndOrderFront:] + 40
	92  QuickLookUI                         0x00000001fe1ae640 -[QLSeamlessDocumentOpener showWindow:contentFrame:withBlock:] + 184
	93  AppKit                              0x00000001933a6a30 -[NSWindowController showWindow:] + 496
	94  SwiftUI                             0x00000001bb1d81cc OUTLINED_FUNCTION_1 + 18364
	95  SwiftUI                             0x00000001bb1db4a4 OUTLINED_FUNCTION_1 + 31380
	96  SwiftUI                             0x00000001bb210de0 OUTLINED_FUNCTION_0 + 9760
	97  SwiftUI                             0x00000001bb08c474 OUTLINED_FUNCTION_0 + 944
	98  SwiftUI                             0x00000001bbb4d200 OUTLINED_FUNCTION_2 + 900
	99  SwiftUI                             0x00000001ba8c8b78 OUTLINED_FUNCTION_11 + 6464
	100 SwiftUI                             0x00000001ba8c9498 objectdestroyTm + 376
	101 SwiftUI                             0x00000001babcada0 OUTLINED_FUNCTION_2 + 13168
	102 SwiftUI                             0x00000001babcadf4 OUTLINED_FUNCTION_2 + 13252
	103 AppKit                              0x0000000193380464 -[NSApplication(NSResponder) sendAction:to:from:] + 460
	104 AppKit                              0x000000019344a698 -[NSMenuItem _corePerformAction] + 372
	105 AppKit                              0x00000001939da874 _NSMenuPerformActionWithHighlighting + 152
	106 AppKit                              0x0000000193482b34 -[NSMenu performActionForItemAtIndex:] + 200
	107 AppKit                              0x0000000193482a54 -[NSMenu _internalPerformActionForItemAtIndex:] + 76
	108 AppKit                              0x00000001939d15e0 +[NSCocoaMenuImpl _performActionForMenuItem:] + 60
	109 AppKit                              0x00000001937fd3a8 -[NSMenuTrackingSession _performPostTrackingDismissalActions] + 264
	110 AppKit                              0x00000001937fd154 -[NSMenuTrackingSession startRunningMenuEventLoop:] + 684
	111 AppKit                              0x0000000193f2e85c -[NSContextMenuTrackingSession startMonitoringEventsInMode:] + 76
	112 AppKit                              0x00000001937b6508 +[NSContextMenuImpl presentPopup:fromView:withContext:animated:] + 640
	113 AppKit                              0x00000001939dbfc4 _NSPopUpMenu + 1988
	114 AppKit                              0x00000001939db72c -[NSCocoaMenuImpl popUpMenu:atLocation:width:view:selectedItemIndexProvider:font:positioningRect:flags:options:] + 432
	115 AppKit                              0x00000001939db554 -[NSCocoaMenuImpl popUpMenu:atLocation:width:forView:withSelectedItem:withFont:withFlags:withOptions:] + 96
	116 AppKit                              0x00000001939a4488 +[NSStatusBarButtonCell popUpStatusBarMenu:ofItem:ofBar:inRect:ofView:withEvent:] + 320
	117 AppKit                              0x00000001939a3880 -[NSStatusBarButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 168
	118 AppKit                              0x000000019337c3c0 -[NSControl mouseDown:] + 448
	119 AppKit                              0x000000019337b18c -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] + 3472
	120 AppKit                              0x0000000193306690 -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 364
	121 AppKit                              0x0000000193306350 -[NSWindow(NSEventRouting) sendEvent:] + 284
	122 AppKit                              0x00000001939a5a9c -[NSStatusBarWindow sendEvent:] + 704
	123 AppKit                              0x00000001939aff30 -[NSApplication(NSEventRouting) sendEvent:] + 1604
	124 AppKit                              0x0000000193602110 -[NSApplication _handleEvent:] + 60
	125 AppKit                              0x00000001931ce124 -[NSApplication run] + 512
	126 AppKit                              0x00000001931a53cc NSApplicationMain + 880
	127 SwiftUI                             0x00000001ba30cf80 OUTLINED_FUNCTION_12 + 15824
	128 SwiftUI                             0x00000001bab9d30c OUTLINED_FUNCTION_3 + 196
	129 SwiftUI                             0x00000001bafe34f0 OUTLINED_FUNCTION_1 + 152
	130 Easydict-debug                      0x0000000104a164c8 $s8Easydict0A17CmpatibilityEntryO4mainyyFZ + 100
	131 Easydict-debug                      0x0000000104a165d8 $s8Easydict0A17CmpatibilityEntryO5$mainyyFZ + 12
	132 Easydict-debug                      0x0000000104a18694 main + 12
	133 dyld                                0x000000018f5a90e0 start + 2360
)
close floating window: <EZFixedQueryWindow: 0x10c809d10>
[EZEventMonitor ● 523 ● I] -[EZEventMonitor getSelectedTextByAccessibility:] ● --> Accessibility success, getText: *** -[__NSFrozenArrayM objectAtIndex:]: index 13 beyond bounds [0 .. 12]
(
	0   CoreFoundation                      0x000000018fa76800 __exceptionPreprocess + 176
	1   libobjc.A.dylib                     0x000000018f56deb4 objc_exception_throw + 60
	2   CoreFoundation                      0x000000018f9e8be0 _CFURLGetObjectInformationNoIO + 0
	3   libswiftCore.dylib                  0x000000019f1a9250 $ss12_ArrayBufferV19_getElementSlowPathyyXlSiF + 196
	4   libswiftCore.dylib                  0x000000019f1b02e8 $sSayxSicir + 304
	5   libswiftCore.dylib                  0x000000019f1b0184 $sSayxGSlsSly7ElementQz5IndexQzcirTW + 68
	6   SwiftUI                             0x00000001bbaee600 OUTLINED_FUNCTION_24 + 6872
	7   SwiftUI                             0x00000001bbaed0d4 OUTLINED_FUNCTION_24 + 1452
	8   SwiftUI                             0x00000001bbaee518 OUTLINED_FUNCTION_24 + 6640
	9   Easydict-debug                      0x0000000104a27480 $s8Easydict10ServiceTabV11serviceListQrvg7SwiftUI4ViewPAEE16listRowSeparator_5edgesQrAE10VisibilityO_AE12VerticalEdgeO3SetVtFQOyAE07DynamicH7ContentPAEE6onMove7performQry10Foundation05IndexP0V_SitcSg_tFQOyAE7ForEachVySaySi6offset_So13EZServiceTypea7elementtGSiAgEE0iJ10BackgroundyQrqd__SgAeFRd__lFQOyAgEE0S10TapGesture5countASQrSi_yyctFQOyAgEE12contentShape_6eoFillQrqd___SbtAE5ShapeRd__lFQOyAgEE3tagyQrqd__SHRd__lFQOyAA0b4ItemH0V_SiQo__AE9RectangleVQo__Qo__AE5ColorVQo_G_Qo__Qo_yXEfU_A20_Si_A0_tcfU_ + 632
	10  Easydict-debug                      0x0000000104a2c1f8 $s8Easydict10ServiceTabV11serviceListQrvg7SwiftUI4ViewPAEE16listRowSeparator_5edgesQrAE10VisibilityO_AE12VerticalEdgeO3SetVtFQOyAE07DynamicH7ContentPAEE6onMove7performQry10Foundation05IndexP0V_SitcSg_tFQOyAE7ForEachVySaySi6offset_So13EZServiceTypea7elementtGSiAgEE0iJ10BackgroundyQrqd__SgAeFRd__lFQOyAgEE0S10TapGesture5countASQrSi_yyctFQOyAgEE12contentShape_6eoFillQrqd___SbtAE5ShapeRd__lFQOyAgEE3tagyQrqd__SHRd__lFQOyAA0b4ItemH0V_SiQo__AE9RectangleVQo__Qo__AE5ColorVQo_G_Qo__Qo_yXEfU_A20_Si_A0_tcfU_TA + 16
	11  Easydict-debug                      0x0000000104a28188 $sSiSo13EZServiceTypea7SwiftUI15ModifiedContentVyAC4ViewPACE12onTapGesture5count7performQrSi_yyctFQOyAEyAEy8Easydict011ServiceItemG0VAC21_TraitWritingModifierVyAC08TagValueP3KeyVySiGGGAC01_f5ShapeR0VyAC9RectangleVGG_Qo_AOyAC017ListRowBackgroundpU0VGGIegygr_Si6offset_AB7elementtA3_Iegnr_TR + 36
	12  Easydict-debug                      0x0000000104a2c250 $sSiSo13EZServiceTypea7SwiftUI15ModifiedContentVyAC4ViewPACE12onTapGesture5count7performQrSi_yyctFQOyAEyAEy8Easydict011ServiceItemG0VAC21_TraitWritingModifierVyAC08TagValueP3KeyVySiGGGAC01_f5ShapeR0VyAC9RectangleVGG_Qo_AOyAC017ListRowBackgroundpU0VGGIegygr_Si6offset_AB7elementtA3_Iegnr_TRTA + 20
	13  SwiftUI                             0x00000001bb4ece84 OUTLINED_FUNCTION_52 + 18912
	14  SwiftUI                             0x00000001bb4f7734 OUTLINED_FUNCTION_52 + 62096
	15  SwiftUI                             0x00000001babe918c OUTLINED_FUNCTION_58 + 56
	16  SwiftUI                             0x00000001bb4f7754 OUTLINED_FUNCTION_52 + 62128
	17  libswiftCore.dylib                  0x000000019f1df6a4 $ss24withUnsafeMutablePointer2to_q_xz_q_SpyxGKXEtKr0_lF + 28
	18  SwiftUI                             0x00000001bb4ec25c OUTLINED_FUNCTION_52 + 15800
	19  SwiftUI                             0x00000001bb4eeabc OUTLINED_FUNCTION_52 + 26136
	20  SwiftUI                             0x00000001bb4f04d0 OUTLINED_FUNCTION_52 + 32812
	21  SwiftUI                             0x00000001bb4f4c84 OUTLINED_FUNCTION_52 + 51168
	22  SwiftUI                             0x00000001bb4f50c0 OUTLINED_FUNCTION_52 + 52252
	23  SwiftUI                             0x00000001bbcffba8 OUTLINED_FUNCTION_18 + 13980
	24  SwiftUI                             0x00000001bbcfc964 OUTLINED_FUNCTION_18 + 1112
	25  SwiftUI                             0x00000001bac1425c OUTLINED_FUNCTION_3 + 5196
	26  SwiftUI                             0x00000001bac15418 OUTLINED_FUNCTION_3 + 9736
	27  SwiftUI                             0x00000001bac1561c OUTLINED_FUNCTION_3 + 10252
	28  SwiftUI                             0x00000001ba9e3fbc OUTLINED_FUNCTION_1 + 3692
	29  SwiftUI                             0x00000001baa200f0 OUTLINED_FUNCTION_64 + 5120
	30  SwiftUI                             0x00000001baa20178 OUTLINED_FUNCTION_64 + 5256
	31  AppKit                              0x0000000193309b60 -[NSTableView _delegate_isGroupRow:] + 80
	32  AppKit                              0x0000000193d9c3e8 -[NSTableRowData _cacheGroupRowIndexesWithinRange:] + 100
	33  AppKit                              0x0000000193d9c340 -[NSTableRowData isGroupRowIndex:] + 144
	34  AppKit                              0x0000000193896250 -[NSTableView _spacingPrecedingRow:] + 44
	35  AppKit                              0x0000000193d10fbc -[NSTableRowHeightData _cacheRowSpansInRange:heightProvider:] + 588
	36  AppKit                              0x0000000193d0f534 -[NSTableRowHeightData _cacheRowSpansInRange:] + 100
	37  Foundation                          0x0000000190ae5570 __NSINDEXSET_IS_CALLING_OUT_TO_A_RANGE_BLOCK__ + 24
	38  Foundation                          0x00000001911faa94 __NSIndexSetEnumerateBitfield + 408
	39  AppKit                              0x0000000193d111fc -[NSTableRowHeightData _estimatedSpanForRow:cacheIfNecessary:] + 256
	40  AppKit                              0x0000000193d0e6e4 -[NSTableRowHeightData _estimatedSpanForRowsInRange:] + 48
	41  AppKit                              0x0000000193d0e818 -[NSTableRowHeightData computeTotalRowsSpan] + 68
	42  AppKit                              0x0000000193d0ea24 -[NSTableRowHeightData _estimatedRowAtOffset:] + 36
	43  AppKit                              0x0000000193263384 -[NSTableView rowAtPoint:] + 204
	44  AppKit                              0x0000000193d9a360 -[NSTableRowData _keepTopRowStableAtLeastOnce:andDoWorkUntilDone:] + 192
	45  AppKit                              0x0000000193271fc8 -[NSTableRowData _updateVisibleViewsBasedOnUpdateItems] + 1300
	46  AppKit                              0x000000019325a780 -[NSTableRowData endUpdates] + 328
	47  AppKit                              0x000000019325a5f4 -[NSTableView _endUpdateWithTile:] + 108
	48  SwiftUI                             0x00000001baa1f324 OUTLINED_FUNCTION_64 + 1588
	49  SwiftUI                             0x00000001bb3d53cc OUTLINED_FUNCTION_23 + 65416
	50  SwiftUI                             0x00000001bb3d4570 OUTLINED_FUNCTION_23 + 61740
	51  SwiftUI                             0x00000001bb3d39e8 OUTLINED_FUNCTION_23 + 58788
	52  SwiftUI                             0x00000001bb3d2418 OUTLINED_FUNCTION_23 + 53204
	53  SwiftUI                             0x00000001baaa6414 OUTLINED_FUNCTION_0 + 16392
	54  SwiftUI                             0x00000001baaab64c __swift_memcpy216_8 + 904
	55  SwiftUI                             0x00000001bb8cf4b0 OUTLINED_FUNCTION_1 + 9376
	56  SwiftUI                             0x00000001baaa60e0 OUTLINED_FUNCTION_0 + 15572
	57  SwiftUI                             0x00000001bb544888 __swift_allocate_boxed_opaque_existential_1Tm + 268
	58  AttributeGraph                      0x00000001bc071790 _ZN2AG5Graph11UpdateStack6updateEv + 512
	59  AttributeGraph                      0x00000001bc071fcc _ZN2AG5Graph16update_attributeENS_4data3ptrINS_4NodeEEEj + 424
	60  AttributeGraph                      0x00000001bc080814 _ZN2AG8Subgraph6updateEj + 848
	61  SwiftUI                             0x00000001bb97e5a4 OUTLINED_FUNCTION_8 + 12300
	62  SwiftUI                             0x00000001bba539cc OUTLINED_FUNCTION_2 + 21436
	63  SwiftUI                             0x00000001bba50670 OUTLINED_FUNCTION_2 + 8288
	64  SwiftUI                             0x00000001bb0da194 OUTLINED_FUNCTION_0 + 59928
	65  SwiftUI                             0x00000001bb0e5768 OUTLINED_FUNCTION_0 + 106476
	66  SwiftUI                             0x00000001bb0da628 OUTLINED_FUNCTION_0 + 61100
	67  AppKit                              0x00000001931f1a98 +[NSAnimationContext runAnimationGroup:] + 56
	68  SwiftUI                             0x00000001bb0da070 OUTLINED_FUNCTION_0 + 59636
	69  SwiftUI                             0x00000001bb0da668 OUTLINED_FUNCTION_0 + 61164
	70  AppKit                              0x0000000193c686b0 ___NSViewLayout_block_invoke + 592
	71  AppKit                              0x000000019321cab0 NSPerformVisuallyAtomicChange + 108
	72  AppKit                              0x00000001932215e0 _NSViewLayout + 96
	73  AppKit                              0x0000000193c5ec88 __36-[NSView _layoutSubtreeWithOldSize:]_block_invoke + 364
	74  AppKit                              0x000000019321cab0 NSPerformVisuallyAtomicChange + 108
	75  AppKit                              0x0000000193221574 -[NSView _layoutSubtreeWithOldSize:] + 100
	76  AppKit                              0x0000000193c5edcc __36-[NSView _layoutSubtreeWithOldSize:]_block_invoke + 688
	77  AppKit                              0x000000019321cab0 NSPerformVisuallyAtomicChange + 108
	78  AppKit                              0x0000000193221574 -[NSView _layoutSubtreeWithOldSize:] + 100
	79  AppKit                              0x0000000193c5f770 __56-[NSView _layoutSubtreeIfNeededAndAllowTemporaryEngine:]_block_invoke + 796
	80  AppKit                              0x000000019321cab0 NSPerformVisuallyAtomicChange + 108
	81  AppKit                              0x0000000193221134 -[NSView _layoutSubtreeIfNeededAndAllowTemporaryEngine:] + 100
	82  AppKit                              0x000000019321cab0 NSPerformVisuallyAtomicChange + 108
	83  AppKit                              0x00000001932210c4 -[NSView layoutSubtreeIfNeeded] + 96
	84  AppKit                              0x0000000193f0a528 -[NSWindow(NSConstraintBasedLayoutInternal) _layoutViewTree] + 104
	85  AppKit                              0x0000000193f0a6b0 -[NSWindow(NSConstraintBasedLayoutInternal) layoutIfNeeded] + 240
	86  AppKit                              0x00000001932d80b0 -[NSWindow _setUpFirstResponderBeforeBecomingVisible] + 68
	87  AppKit                              0x00000001932d7598 -[NSWindow _doWindowWillBeVisibleAsSheet:] + 108
	88  AppKit                              0x0000000193c7bd14 -[NSWindow _reallyDoOrderWindowAboveOrBelow:] + 1028
	89  AppKit                              0x0000000193c7c8c4 -[NSWindow _reallyDoOrderWindow:] + 64
	90  AppKit                              0x0000000193c7cb24 -[NSWindow _doOrderWindow:] + 264
	91  AppKit                              0x00000001932cd774 -[NSWindow makeKeyAndOrderFront:] + 40
	92  QuickLookUI                         0x00000001fe1ae640 -[QLSeamlessDocumentOpener showWindow:contentFrame:withBlock:] + 184
	93  AppKit                              0x00000001933a6a30 -[NSWindowController showWindow:] + 496
	94  SwiftUI                             0x00000001bb1d81cc OUTLINED_FUNCTION_1 + 18364
	95  SwiftUI                             0x00000001bb1db4a4 OUTLINED_FUNCTION_1 + 31380
	96  SwiftUI                             0x00000001bb210de0 OUTLINED_FUNCTION_0 + 9760
	97  SwiftUI                             0x00000001bb08c474 OUTLINED_FUNCTION_0 + 944
	98  SwiftUI                             0x00000001bbb4d200 OUTLINED_FUNCTION_2 + 900
	99  SwiftUI                             0x00000001ba8c8b78 OUTLINED_FUNCTION_11 + 6464
	100 SwiftUI                             0x00000001ba8c9498 objectdestroyTm + 376
	101 SwiftUI                             0x00000001babcada0 OUTLINED_FUNCTION_2 + 13168
	102 SwiftUI                             0x00000001babcadf4 OUTLINED_FUNCTION_2 + 13252
	103 AppKit                              0x0000000193380464 -[NSApplication(NSResponder) sendAction:to:from:] + 460
	104 AppKit                              0x000000019344a698 -[NSMenuItem _corePerformAction] + 372
	105 AppKit                              0x00000001939da874 _NSMenuPerformActionWithHighlighting + 152
	106 AppKit                              0x0000000193482b34 -[NSMenu performActionForItemAtIndex:] + 200
	107 AppKit                              0x0000000193482a54 -[NSMenu _internalPerformActionForItemAtIndex:] + 76
	108 AppKit                              0x00000001939d15e0 +[NSCocoaMenuImpl _performActionForMenuItem:] + 60
	109 AppKit                              0x00000001937fd3a8 -[NSMenuTrackingSession _performPostTrackingDismissalActions] + 264
	110 AppKit                              0x00000001937fd154 -[NSMenuTrackingSession startRunningMenuEventLoop:] + 684
	111 AppKit                              0x0000000193f2e85c -[NSContextMenuTrackingSession startMonitoringEventsInMode:] + 76
	112 AppKit                              0x00000001937b6508 +[NSContextMenuImpl presentPopup:fromView:withContext:animated:] + 640
	113 AppKit                              0x00000001939dbfc4 _NSPopUpMenu + 1988
	114 AppKit                              0x00000001939db72c -[NSCocoaMenuImpl popUpMenu:atLocation:width:view:selectedItemIndexProvider:font:positioningRect:flags:options:] + 432
	115 AppKit                              0x00000001939db554 -[NSCocoaMenuImpl popUpMenu:atLocation:width:forView:withSelectedItem:withFont:withFlags:withOptions:] + 96
	116 AppKit                              0x00000001939a4488 +[NSStatusBarButtonCell popUpStatusBarMenu:ofItem:ofBar:inRect:ofView:withEvent:] + 320
	117 AppKit                              0x00000001939a3880 -[NSStatusBarButtonCell trackMouse:inRect:ofView:untilMouseUp:] + 168
	118 AppKit                              0x000000019337c3c0 -[NSControl mouseDown:] + 448
	119 AppKit                              0x000000019337b18c -[NSWindow(NSEventRouting) _handleMouseDownEvent:isDelayedEvent:] + 3472
	120 AppKit                              0x0000000193306690 -[NSWindow(NSEventRouting) _reallySendEvent:isDelayedEvent:] + 364
	121 AppKit                              0x0000000193306350 -[NSWindow(NSEventRouting) sendEvent:] + 284
	122 AppKit                              0x00000001939a5a9c -[NSStatusBarWindow sendEvent:] + 704
	123 AppKit                              0x00000001939aff30 -[NSApplication(NSEventRouting) sendEvent:] + 1604
	124 AppKit                              0x0000000193602110 -[NSApplication _handleEvent:] + 60
	125 AppKit                              0x00000001931ce124 -[NSApplication run] + 512
	126 AppKit                              0x00000001931a53cc NSApplicationMain + 880
	127 SwiftUI                             0x00000001ba30cf80 OUTLINED_FUNCTION_12 + 15824
	128 SwiftUI                             0x00000001bab9d30c OUTLINED_FUNCTION_3 + 196
	129 SwiftUI                             0x00000001bafe34f0 OUTLINED_FUNCTION_1 + 152
	130 Easydict-debug                      0x0000000104a164c8 $s8Easydict0A17CmpatibilityEntryO4mainyyFZ + 100
	131 Easydict-debug                      0x0000000104a165d8 $s8Easydict0A17CmpatibilityEntryO5$mainyyFZ + 12
	132 Easydict-debug                      0x0000000104a18694 main + 12
	133 dyld                                0x000000018f5a90e0 start + 2360
)
```

Reply:
-1. zip/enumerated are both unacceptable code IMO. The id: "\.offset" is wrong.

Reply:
```
ForEach(Array(zip(serviceTypes.indices, serviceTypes)), id: \.0) { index, _ in
```
How about use like this?

Reply:
I found the problem, it is due to the inconsistency between the number of serviceTypes and services, I debugged and found that the serviceTypes count is 14 and services is 13.

You can cut to this [Gemini](https://github.com/tisfeng/Easydict/pull/297) branch and try, since the new service type is written to the local serviceTypes store, and the new service class is not in the current code.

```swift
serviceTypes = EZLocalStorage.shared().allServiceTypes(windowType)
services = EZLocalStorage.shared().allServices(windowType)
```

Reply:
All right, so after the new Gemini service is introduced, it would be ok?

Reply:
Yes, `services` is always safe.